### PR TITLE
Adding MediaPlayer bundle factory

### PR DESCRIPTION
**Pull request by tchakabam**
https://www.pivotaltracker.com/projects/1597593/stories/120758659

only remaining question: is this kind of replacement of the Dash.js factory what we would like to deliver or do we actually want to extend Dash.js properly?

Probably this already serves the purpose well I could imagine (see `bundle.html` example file)

### CODE REVIEW

File: lib/DashjsMediaPlayerBundle.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import dashjs from 'dashjs';
+
+function MediaPlayer(p2pConfig) {
+  const liveDelay = 30;
+  let player = dashjs.MediaPlayer().create();
+
+  new DashjsWrapper(player, p2pConfig.videoElement, p2pConfig, liveDelay);
```
Comment (ruslandinov) [inline lib/DashjsMediaPlayerBundle.js]: videoElement in p2pConfig? Why?

Comment (tchakabam) [inline lib/DashjsMediaPlayerBundle.js]: because i thought that we should rather pass anything that we need in one config object, especially since this constructor is a modified form of the original.

By the way, I think we don't need the video-element in the peer-agent version used by hlsjs-wrapper (v4). so we might not need it anymore here as well?

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1 @@
+dashjs/
```
Comment (maxbok) [inline .npmignore]: You should also put `toolkit`, `bridge`, `dist/bundle` and `.ruby-version` in it.

File: circle.yml [hunk 3]
```diff
@@ -36,21 +35,20 @@ deployment:
         branch: /^(?!(?:dev|master)$).+$/
         commands:
             # Upload dist
-            # - grunt browserify:bundle
-            - grunt browserify:wrapper
+            - npm run build
 
-            # - toolkit/add_banner.rb --file dist/bundle/$DIST_BUNDLE --deploy_env $CIRCLE_BRANCH
+            - toolkit/add_banner.rb --file dist/bundle/$DIST_BUNDLE --deploy_env $CIRCLE_BRANCH
             - toolkit/add_banner.rb --file dist/wrapper/$DIST_WRAPPER --deploy_env $CIRCLE_BRANCH
```
Comment (maxbok) [inline circle.yml]: You should add the `--dependencies` and `name` options here.
Like we do in `hls-wrapper`: https://github.com/streamroot/hlsjs-wrapper/blob/dev/circle.yml#L43-L49

`--dependencies` will display the version of the listed dependencies.
`--name` will override the module name. Useful for the `bundle`.

Comment (tchakabam) [inline circle.yml]: i see duplication, aaarggg 

we should really make a script for all the various calls to `add_banner.rb` i.e to sum up what happens in CI maybe

**Comment by maxbok** [comment]
:+1:

File: example/bundle.html [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
-                    <option value="http://wowza.streamroot.io/vodOriginSite/tears_of_steel720p.mp4/manifest.mpd">ToS (VoD, 1 video)</option>
-                    <option value="http://wowza.streamroot.io/liveorigin/stream4/manifest.mpd">ToS (Live DVR, 1 video)</option>
-                    <option value="http://wowza.streamroot.io/liveorigin/Sample-live.smil/manifest.mpd">Sintel (Live DVR, 3 video)</option>
-                    <option value="http://wowza.streamroot.io/vodOriginSite/Sample-vod.smil/manifest.mpd">Sintel (VoD, 3 video)</option>
+                    <option value="http://akamai.streamroot.edgesuite.net/vodorigin/sintel.mp4/manifest.mpd">Sintel VOD (segment timeline | 1 video & 1 audio)</option>
+                    <option value="http://vm2.dashif.org/livesim/testpic_2s/Manifest.mpd">Live 2s seg(number template | 1 video & 1 audio)</option>
+                    <option value="http://media.streamroot.io/dash/examples/snow15/snowboard1_720p_dash.mpd">Snowboard 1:30(segment base | 3 video & 1 audio)</option>
+                    <option value="http://dash.edgesuite.net/envivio/Envivio-dash2/manifest.mpd">Envivio 3:13(number template | 9 video & 1 audio)</option>
```
Comment (ruslandinov) [inline example/bundle.html]: 3 weeks ago we(me and Remi) decided to use our own(Streamroot) demo streams for demo pages.
Why did you change these url?

**Comment by ruslandinov** [comment]
The rest looks good for me 👍

File: example/bundle.html (hunk 4 above)
Comment (tchakabam) [inline example/bundle.html]: did i? actually i just used the URLs in the "local" demo page.

i guess these were only in the plain demo page. i will put back the `wowza.streamroot.io` ones.

thanks for noticing this!!

**Comment by tchakabam** [comment]
@ruslandinov fixed the URLs to be ours again!

**Comment by tchakabam** [comment]
I will update the README, then merge
